_posts/2022-04-15-the-comprehensive-guide-to-elixirs-for-comprehension.md

```diff
@@ -0,0 +1,673 @@
+
+Let's take a look at some examples.
+
+#### List
+
+Using a list as the `:into` actually doesn't change the behavior at all!
+
+
+```elixir
+employees = [
+  %{
+    name: "Eric",
+    status: :active,
+    hobbies: [%{name: "Text Adventures", type: :gaming}, %{name: "Chickens", type: :animals}]
+  },
+  %{
+    name: "Mitch",
+    status: :former,
+    hobbies: [%{name: "Woodworking", type: :making}, %{name: "Homebrewing", type: :making}]
+  },
+  %{
+    name: "Greg",
+    status: :active,
+    hobbies: [
+      %{name: "Dungeons & Dragons", type: :gaming},
+      %{name: "Woodworking", type: :making}
+    ]
+  }
+]
+
+for employee <- employees,
+    employee.status == :active,
+    hobby <- employee.hobbies,
+    hobby.type == :gaming,
+    into: [] do
+  {employee.name, hobby}
+end
+
+# [
+#   {"Eric", %{name: "Text Adventures", type: :gaming}},
+#   {"Greg", %{name: "Dungeons & Dragons", type: :gaming}}
+# ]
+```
+
+#### Map
+
+But if we use a map, we can see that it pushes each key/value pair into the map that you pass for the option. Usually I use this with an empty map (`%{}` or `Map.new()`), but let's look at an example using a non-empty map.
+
+```elixir
+employees = [
+  %{
+    name: "Eric",
+    status: :active,
+    hobbies: [%{name: "Text Adventures", type: :gaming}, %{name: "Chickens", type: :animals}]
+  },
+  %{
+    name: "Mitch",
+    status: :former,
+    hobbies: [%{name: "Woodworking", type: :making}, %{name: "Homebrewing", type: :making}]
+  },
+  %{
+    name: "Greg",
+    status: :active,
+    hobbies: [
+      %{name: "Dungeons & Dragons", type: :gaming},
+      %{name: "Woodworking", type: :making}
+    ]
+  }
+]
+
+base_map = %{
+  "Mitch" => %{
+    name: "Reading",
+    type: :learning
+  },
+  "Greg" => %{
+    name: "Traveling",
+    type: :expensive
+  }
+}
+
+for employee <- employees,
+    employee.status == :active,
+    hobby <- employee.hobbies,
+    hobby.type == :gaming,
+    into: base_map do
+  {employee.name, hobby}
+end
+
+# %{
+#   "Eric" => %{name: "Text Adventures", type: :gaming},
+#   "Greg" => %{name: "Dungeons & Dragons", type: :gaming},
+#   "Mitch" => %{name: "Reading", type: :learning}
+# }
+```
+
+Here we can observe three things.
+
+- The comprehension evaluates to a map.
+- The `"Mitch"` key and its value were preserved in the final output.
+- The `"Greg"` key's value in the `base_map` was overwritten by the value yielded during the comprehension with the same key. If our comprehension were to have returned multiple key/value pairs with identical keys, the last one would have won.
+
+This option is very useful for transforming maps, since iterating over a map with an `Enum` function turns it into a list of 2-tuples, you always need to pipe the return value into `Enum.into/2` or `Map.new/1`.
```

Comment:
```suggestion
This option is very useful for transforming maps. Since iterating over a map with an `Enum` function turns it into a list of 2-tuples, you always need to pipe the return value into `Enum.into/2` or `Map.new/1`.
```